### PR TITLE
use already exist func replace some redundance code

### LINKED ISSUE #55720: `kubectl get all` lists some resource types twice
**Is this a BUG REPORT or FEATURE REQUEST?**:

/kind bug

**What happened**:

`kubectl get all` prints the replica set section twice.

**What you expected to happen**:

It should not print dupes.

**How to reproduce it (as minimally and precisely as possible)**:

```shell
$ kubectl get all -l app=myapp
NAME                           DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deploy/myapp        4         4         4            4           169d

NAME                                  DESIRED   CURRENT   READY     AGE
rs/myapp-1031441472        0         0         0         159d
rs/myapp-1066912010        0         0         0         159d
rs/myapp-1627954893        0         0         0         140d
rs/myapp-1660980575        0         0         0         154d
rs/myapp-2403377231        0         0         0         137d
rs/myapp-2479791910        4         4         4         53d
rs/myapp-3454720083        0         0         0         145d
rs/myapp-3461390017        0         0         0         139d
rs/myapp-3554658816        0         0         0         162d
rs/myapp-3624978688        0         0         0         169d
rs/myapp-378637676         0         0         0         154d
rs/myapp-4151746477        0         0         0         139d
rs/myapp-702975759         0         0         0         159d

NAME                           DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deploy/myapp        4         4         4            4           169d

NAME                                  DESIRED   CURRENT   READY     AGE
rs/myapp-1031441472        0         0         0         159d
rs/myapp-1066912010        0         0         0         159d
rs/myapp-1627954893        0         0         0         140d
rs/myapp-1660980575        0         0         0         154d
rs/myapp-2403377231        0         0         0         137d
rs/myapp-2479791910        4         4         4         53d
rs/myapp-3454720083        0         0         0         145d
rs/myapp-3461390017        0         0         0         139d
rs/myapp-3554658816        0         0         0         162d
rs/myapp-3624978688        0         0         0         169d
rs/myapp-378637676         0         0         0         154d
rs/myapp-4151746477        0         0         0         139d
rs/myapp-702975759         0         0         0         159d

NAME                                        READY     STATUS    RESTARTS   AGE
po/myapp-2479791910-9w7zx        1/1       Running   0          14d
po/myapp-2479791910-k2p44        1/1       Running   0          14d
po/myapp-2479791910-nb5lj        1/1       Running   0          14d
po/myapp-2479791910-wjmcn        1/1       Running   1          14d

NAME                        TYPE       CLUSTER-IP     EXTERNAL-IP   PORT(S)        AGE
svc/myapp        NodePort   10.7.245.217   <none>        80:32046/TCP   169d
```

Correct:

```
$ kubectl get rs,pod -l app=myapp
NAME                                  DESIRED   CURRENT   READY     AGE
rs/myapp-1031441472        0         0         0         159d
rs/myapp-1066912010        0         0         0         159d
rs/myapp-1627954893        0         0         0         140d
rs/myapp-1660980575        0         0         0         154d
rs/myapp-2403377231        0         0         0         137d
rs/myapp-2479791910        4         4         4         53d
rs/myapp-3454720083        0         0         0         145d
rs/myapp-3461390017        0         0         0         139d
rs/myapp-3554658816        0         0         0         162d
rs/myapp-3624978688        0         0         0         169d
rs/myapp-378637676         0         0         0         154d
rs/myapp-4151746477        0         0         0         139d
rs/myapp-702975759         0         0         0         159d

NAME                                        READY     STATUS    RESTARTS   AGE
po/myapp-2479791910-9w7zx        1/1       Running   0          14d
po/myapp-2479791910-k2p44        1/1       Running   0          14d
po/myapp-2479791910-nb5lj        1/1       Running   0          14d
po/myapp-2479791910-wjmcn        1/1       Running   1          14d
```

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):  1.8.1-gke.1
- Cloud provider or hardware configuration: GKE
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by zjj2wry**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:
cc @lichuqiang 

**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by zjj2wry** [comment]
cc @kubernetes/sig-cli-pr-reviews

File: pkg/kubectl/resource/builder.go [hunk 1]
```diff
@@ -418,12 +418,12 @@ func (b *Builder) ReplaceAliases(input string) string {
 	for _, arg := range strings.Split(input, ",") {
 		if resources, ok := b.categoryExpander.Expand(arg); ok {
 			asStrings := []string{}
+			resourceMap := make(map[string]bool)
 			for _, resource := range resources {
-				if len(resource.Group) == 0 {
-					asStrings = append(asStrings, resource.Resource)
-					continue
+				if _, ok := resourceMap[resource.Resource]; !ok {
```
Comment (dixudx) [inline pkg/kubectl/resource/builder.go]: Better add some tests for this.

**Comment by liggitt** [comment]
if the same resource appears in multiple groups, from the CLI's perspective, it is distinct. ('deployments' in group 'a' are not inherently related to 'deployments' in group 'b')

if both include themselves in the 'all' alias, I'd expect both to appear

**Comment by dixudx** [comment]
> if the same resource appears in multiple groups, from the CLI's perspective, it is distinct. 

Agree. +1

**Comment by zjj2wry** [comment]
We need distinguish them with different group when use `all`?
```
admindeMacBook-Pro-2:kubectl admin$ ./kubectl get all
NAME            DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deployments.extensions/deploy   1         1         1            1           3h

NAME            DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
deployments.extensions/deploy   1         1         1            1           3h
```

**Comment by zjj2wry** [comment]
when i write this PR find this String() func can use directly. will small enhance code...
```
func (gr *GroupResource) String() string {
	if len(gr.Group) == 0 {
		return gr.Resource
	}
	return gr.Resource + "." + gr.Group
}
```

**Comment by zjj2wry** [comment]
seems this already ref issue, avoid misunderstand this, so close this...will add this code in my another PR

**Pull request closed without merging**